backends/tc/backend.cpp

```diff
@@ -590,6 +590,39 @@ void ConvertToBackendIR::updatePnaDirectCounter(const IR::P4Table *t, IR::TCTabl
     }
 }
 
+void ConvertToBackendIR::updatePnaDirectMeter(const IR::P4Table *t, IR::TCTable *tabledef,
+                                              unsigned tentries) {
+    cstring propertyName = "pna_direct_meter"_cs;
+    auto property = t->properties->getProperty(propertyName);
+    if (property == nullptr) return;
+    auto expr = property->value->to<IR::ExpressionValue>()->expression;
+    auto ctrl = findContext<IR::P4Control>();
+    auto cName = ctrl->name.originalName;
+    auto externInstanceName = cName + "." + expr->toString();
+    tabledef->setDirectMeter(externInstanceName);
+
+    auto externInstance = P4::ExternInstance::resolve(expr, refMap, typeMap);
+    if (!externInstance) {
+        ::P4::error(ErrorType::ERR_INVALID,
+                    "Expected %1% property value for table %2% to resolve to an "
+                    "extern instance: %3%",
+                    propertyName, t->name.originalName, property);
+        return;
+    }
+    for (auto ext : externsInfo) {
+        if (ext.first == externInstance->type->toString()) {
+            auto externDefinition = tcPipeline->getExternDefinition(ext.first);
+            if (externDefinition) {
+                auto extInstDef = ((IR::TCExternInstance *)externDefinition->getExternInstance(
+                    externInstanceName));
+                extInstDef->setExternTableBindable(true);
+                extInstDef->setNumElements(tentries);
+                break;
+            }
+        }
+    }
```

Comment:
no error check/ message when not found?
Silent failure is that expected?

Reply:
It is not a bug, as we are unable to find the extern definition, we didnt attach the number of elements to it.

Reply:
in what use case we will endup here

Reply:
If the direct meter extern attached to table via "pna_direct_meter" is not declared. However in that case, the error will be caught in frontend itself.

Reply:
In that case it has to be assert or an error here

Reply:
addressed.